File: leon-samples/leon-samples-mixed/src/main/scala/io/leon/samples/mixed/Module.scala
/*
 * Copyright (c) 2011 WeigleWilczek and others.
 * 
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
package io.leon.samples.mixed

import io.leon.AbstractLeonConfiguration
import io.leon.persistence.mongo.LeonMongoModule
import io.leon.resources.coffeescript.CoffeeScriptModule
import com.google.inject.Inject
import io.leon.javascript.{JavaScriptProxy, LeonScriptEngine}

class Module extends AbstractLeonConfiguration {

  def config() {
    install(new LeonMongoModule)
    install(new CoffeeScriptModule)
    bind(classOf[ModuleInit]).asEagerSingleton()

    loadFile("/io/leon/samples/mixed/person.js")

    browser("person").linksToServer("person")
    browser("personService").linksToServer("personService")

    server("leon.browser").linksToAllPages("leon")
  }

}

class ModuleInit @Inject() (engine: LeonScriptEngine) {
  engine.put("personService", JavaScriptProxy(engine.rhinoScope, new PersonService))
}

File: leon-samples/angular/crud/src/main/java/com/acme/appname/Module.java
package com.acme.appname;

import io.leon.LeonAppMainModule;
import io.leon.persistence.mongo.LeonMongoModule;

public class Module extends LeonAppMainModule {

    @Override
    protected void config() {
        
        setApplicationName("AcmeAppName");

        install(new LeonMongoModule());

        loadFile("index.server.js");
        exposeJavaScript("/indexService", "indexService");
    }

}
